ios/ble/Utility/TransferHandler.swift

```diff
@@ -67,15 +67,17 @@ class TransferHandler {
 
     private func sendRetryRespChunk(missingChunks: [Int]) {
         for chunkIndex in missingChunks {
-            let chunk = chunker?.getChunkWithIndex(index: chunkIndex)
-            Central.shared.write(serviceUuid: Peripheral.SERVICE_UUID, charUUID: NetworkCharNums.SUBMIT_RESPONSE_CHAR_UUID, data: chunk!)
+            if let chunk = chunker?.getChunkWithIndex(index: chunkIndex) {
+                delegate?.write(serviceUuid: Peripheral.SERVICE_UUID, charUUID: NetworkCharNums.SUBMIT_RESPONSE_CHAR_UUID, data: chunk, withResponse: true)
+            }
+            // checks if no more missing chunks exist on verifier
         }
         sendMessage(message: imessage(msgType: .READ_TRANSMISSION_REPORT, data: nil))
     }
 
     private func requestTransmissionReport() {
         var notifyObj: Data
-        Central.shared.write(serviceUuid: BLEConstants.SERVICE_UUID, charUUID: NetworkCharNums.TRANSFER_REPORT_REQUEST_CHAR_UUID, data: withUnsafeBytes(of: 1.littleEndian) { Data($0) })
+        delegate?.write(serviceUuid: BLEConstants.SERVICE_UUID, charUUID: NetworkCharNums.TRANSFER_REPORT_REQUEST_CHAR_UUID, data: withUnsafeBytes(of: 1.littleEndian) { Data($0) }, withResponse: false)
```

Comment:
This was changed recently to write with response in the last release and there is retry mechanism on top of this.

Reply:
We have added a unique write method which takes in with or without response as a parameter and change based on that.